src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -84,10 +83,30 @@ static void report_store_failure() {
     vm_abort(false);
   }
   log_info(aot, codecache, exit)("Unable to create AOT Code Cache.");
-  AOTAdapterCaching = false;
-  AOTStubCaching = false;
+  AOTCodeCache::disable_caching();
 }
 
+// The sequence of AOT code caching flags and parametters settings.
+//
+// 1. The initial AOT code caching flags setting is done
+// during call to CDSConfig::check_vm_args_consistency().
+//
+// 2. The earliest AOT code state check done in compilationPolicy_init()
+// where we set number of compiler threads for AOT assembly phase.
+//
+// 3. We determine presence of AOT code in AOT Cache in
+// MetaspaceShared::open_static_archive() which is calles
+// after compilationPolicy_init() but before codeCache_init().
+//
+// 4. AOTCodeCache::initialize() is called during universe_init()
+// and does final AOT state and flags settings.
+//
+// 5. Finally AOTCodeCache::init2() is called after universe_init()
+// when all GC settings are finalized.
+
+// Next methods determine which action we do with AOT code depending
+// on phase of AOT process: assembly or production.
+
```

Comment:
This comment and following new AOT functions are added to match leyden/premain branch.